tools/odbc/prepared.cpp

```diff
@@ -1,6 +1,49 @@
 #include "duckdb_odbc.hpp"
 #include "statement_functions.hpp"
 #include "duckdb/main/prepared_statement_data.hpp"
+#include "duckdb/common/types/string_type.hpp"
+#include "duckdb/common/operator/cast_operators.hpp"
+#include "duckdb/common/types/decimal.hpp"
+#include "duckdb/common/exception.hpp"
+
+using duckdb::Decimal;
+using duckdb::hugeint_t;
+using duckdb::idx_t;
+using duckdb::Load;
+using duckdb::string_t;
+using duckdb::Value;
+
+bool IsNumeric(SQLSMALLINT value_type) {
+	switch (value_type) {
+	case SQL_C_TINYINT:
+	case SQL_C_STINYINT:
+	case SQL_C_UTINYINT:
+	case SQL_C_SHORT:
+	case SQL_C_SSHORT:
+	case SQL_C_USHORT:
+	case SQL_C_SLONG:
+	case SQL_C_LONG:
+	case SQL_C_ULONG:
+	case SQL_C_SBIGINT:
+	case SQL_C_UBIGINT:
+	case SQL_C_FLOAT:
+	case SQL_C_DOUBLE:
+	case SQL_NUMERIC:
+		return true;
+	default:
+		return false;
+	}
+}
+
+void ValidateNumeric(int precision, int scale) {
+	// if( !(precision >= 1 && precision <= Decimal::MAX_WIDTH_DECIMAL && scale <= precision) )
+	if (precision < 1 || precision > Decimal::MAX_WIDTH_DECIMAL || scale < 0 || scale > Decimal::MAX_WIDTH_DECIMAL ||
+	    scale > precision) {
+		std::string msg = "Numeric precision: " + std::to_string(precision) + " and scale: " + std::to_string(scale) +
+		                  " not supported.";
+		throw duckdb::NotImplementedException(msg);
```

Comment:
Exceptions should never be thrown in a C-API (like the ODBC). This should return an error message instead.